### PR TITLE
Adding edge tests for API's

### PR DESCRIPTION
**Pull request by Smit1603**

### CODE REVIEW

File: lib/Interpreter/CppInterOp.cpp [hunk 1]
```diff
@@ -422,10 +426,6 @@ namespace Cpp {
       return ND->getNameAsString();
     }
 
-    if (llvm::isa_and_nonnull<TranslationUnitDecl>(D)) {
-      return "";
-    }
-
```
Comment (vgvassilev) [inline lib/Interpreter/CppInterOp.cpp:428]: Why we had to move this?

Comment (Smit1603) [reply lib/Interpreter/CppInterOp.cpp:428]: Actually thats an error from my hand reverting it back to original state

Comment (vgvassilev) [reply lib/Interpreter/CppInterOp.cpp]: Ok, then please undo that change.

File: lib/Interpreter/CppInterOp.cpp [hunk 2]
```diff
@@ -406,8 +406,8 @@ namespace Cpp {
     auto &C = getSema().getASTContext();
     auto *D = (Decl *) klass;
 
-    if (auto *ND = llvm::dyn_cast_or_null<NamedDecl>(D)) {
-      if (auto *TD = llvm::dyn_cast<TagDecl>(ND)) {
+    if (auto* ND = llvm::dyn_cast_or_null<NamedDecl>(D)) {
+      if (auto* TD = llvm::dyn_cast<TagDecl>(ND)) {
```
[the remark this answers is not in the document]
Comment (Smit1603) [reply lib/Interpreter/CppInterOp.cpp]: Done it , Sry for this auto correction in VS-Code

**Review by vgvassilev — APPROVED** [review]
LGTM!

**Pull request merged**